dask_remote/client_base.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+from urllib import parse
+
+from requests import Session, auth
+
+
+# Default connection timeout at 10''
+DEFAULT_TIMEOUT = 10.0
+
+# Default JSON response back
+DEFAULT_HEADERS = {"Accept": "application/json"}
+
+
+class ClientError(Exception):
+    ...
+
+
+class JSONClient:
+    """Base class for JSON clients."""
+
+    def __init__(
+        self, base_url: str, default_timeout: float = None, default_headers: dict = None
+    ) -> None:
+        self._url = base_url
+        self._timeout = default_timeout or DEFAULT_TIMEOUT
+        self._headers = default_headers or DEFAULT_HEADERS
+        self._connect()
```

Comment:
I would rather have a `self.conn` property that gets initialized on first access?